PowerEditor/src/WinControls/DockingWnd/DockingManager.cpp

```diff
@@ -968,12 +970,21 @@ int DockingManager::FindEmptyContainer()
 
         for (size_t iTb = 0, len = vTbData.size(); iTb < len; ++iTb)
         {
-            pArrayPos[vTbData[iTb]->iPrevCont] = TRUE;
+			if ((vTbData[iTb]->iPrevCont < static_cast<int>(dockingContVectorSize)) && (vTbData[iTb]->iPrevCont >= -1))
+			{
+				pArrayPos[vTbData[iTb]->iPrevCont] = TRUE;

```

Comment:
When **vTbData[iTb]->iPrevCont == -1**, it satisfies the condition `vTbData[iTb]->iPrevCont >= -1`, then `pArrayPos[-1] = TRUE;` will make an explosion.

Reply:
@donho 
No, `-1` is OK here.
It is a Docking Manager expected value meaning `no-previous-one` and the original code is prepared for that (I only added a comment there to explain better that code meaning):
https://github.com/xomx/notepad-plus-plus/blob/ceead7526bb120bee465531f26fd5cb0de347209/PowerEditor/src/WinControls/DockingWnd/DockingManager.cpp#L958

Reply:
pArrayPos is a pointer of BOOL which points to the 2nd position of a BOOL array.
So I don't understand why `pArrayPos[-1]` won't cause the runtime error.

Reply:
@donho 

![npp-pArrayPos-minus-one-index-area](https://github.com/user-attachments/assets/ec4ab9a4-9e78-464b-985f-1ae95c4507d5)

Reply:
Holly S... I didn't see that coming! (Docking system was done about 18 years ago, by a contributor)
Surely it's not a good practice, but do we have any better way for it?

Reply:
BTW, I thought index `-1` in the `[` `]` will be translated to an unsigned int (size_t?) on the runtime?

Reply:
OK, it's about the **POINTER** but not **ARRAY**
https://stackoverflow.com/questions/47170740/c-negative-array-index